src/util/simplify_expr_int.cpp

```diff
@@ -1572,9 +1572,12 @@ simplify_exprt::resultt<> simplify_exprt::simplify_inequality_rhs_is_constant(
     }
 
     // very special case for pointers
-    if(expr.id()==ID_equal &&
-       expr.op1().is_constant() &&
-       expr.op1().get(ID_value)==ID_NULL)
+    if(expr.id() != ID_equal)
+      return unchanged(expr);
+
+    const constant_exprt &op1_constant = to_constant_expr(expr.op1());
```

Comment:
Is it possible that this transfer function throws? Say, if `expr.op1()` is not a `constant_exprt`?

Before this change, a property was checked and there was no transfer function application, so I'm slightly worried that this could result in invariants at runtime in some cases.

Reply:
This previous check actually was unnecessary: the function checks for this earlier on already.